.gitlab/llnl-ci-pip.yml

```diff
@@ -0,0 +1,98 @@
+stages:
+  - env-setup
+  - test
+  - cleanup
+
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks $SDK_WORK_DIR/pip || true
+    - chmod -fR 02770 $SDK_WORK_DIR/pip    || true
+
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks $SDK_WORK_DIR/pip || true
+    - chmod -fR 02770 $SDK_WORK_DIR/pip    || true
+
+pip-env-setup:
+  stage: env-setup
+  parallel:
+    matrix:
+      - HOST : [ "quartz", "ruby"]
+        RUNNER_TYPE: "shell"
+      # - HOST : [ "lassen" ]
+      #   ENVIRONMENT: [ "BATCH" ]
+      #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - mkdir -p ${SDK_WORK_DIR}/pip/
+    - cd ${SDK_WORK_DIR}/pip/
+    - python3 -m venv ${HOST}-venv
+    - cd ${HOST}-venv
+    - source bin/activate
+    - pip install --upgrade pip
```

Comment:
important to have these commands
```
pip install -U pip setuptools wheel
pip cache purge
```

Reply:
`pip cache purge` could be part of cleanup stage for pip

Reply:
Done